pkg/types/urlsmap.go

```diff
@@ -27,15 +26,24 @@ type URLsMap map[string]URLs
 // which consists of discovery-formatted names-to-URLs, like:
 // mach0=http://1.1.1.1:2380,mach0=http://2.2.2.2::2380,mach1=http://3.3.3.3:2380,mach2=http://4.4.4.4:2380
 func NewURLsMap(s string) (URLsMap, error) {
-	cl := URLsMap{}
-	v, err := url.ParseQuery(strings.Replace(s, ",", "&", -1))
-	if err != nil {
-		return nil, err
-	}
-	for name, urls := range v {
-		if len(urls) == 0 || urls[0] == "" {
+	urlsm := make(map[string][]string)
```

Comment:
can we simply copy https://golang.org/src/net/url/url.go?s=18971:19022#L680 and remove the unnecessary parts?